metricproviders/kayenta/kayenta.go

```diff
@@ -0,0 +1,202 @@
+		name := s.Name
+		controlScope, err := json.Marshal(s.ControlScope)
+		if err != nil {
+			return metricutil.MarkMeasurementError(newMeasurement, err)
+		}
+		experimentScope, err := json.Marshal(s.ExperimentScope)
+		if err != nil {
+			return metricutil.MarkMeasurementError(newMeasurement, err)
+		}
+		scopes = "\"" + name +  "\":" + string(controlScope) + "," + string(experimentScope)
+		if i < (len(metric.Provider.Kayenta.Scopes) - 1) {
+			scopes = scopes + ","
+		}
+
+	}
+
+	jobPayLoad = strings.Replace(jobPayLoad, "{{inputs.scopes}}", scopes, 1)
+
+	jsonValue, err := templateutil.ResolveArgs(jobPayLoad, run.Spec.Arguments)
+	if err != nil {
+		return metricutil.MarkMeasurementError(newMeasurement, err)
+	}
+
+	response, err := p.client.Post(jobURL, "application/json", bytes.NewBuffer([]byte(jsonValue)))
+	if err != nil || response.Body == nil || response.StatusCode > 300 {
+		if err == nil {
+			err := errors.New("Invalid Response")
+			return metricutil.MarkMeasurementError(newMeasurement, err)
+		}
+		return metricutil.MarkMeasurementError(newMeasurement, err)
+	} else {
+		data, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return metricutil.MarkMeasurementError(newMeasurement, err)
+		}
+		var dat map[string]interface{}
+		if err := json.Unmarshal(data, &dat); err != nil {
+			return metricutil.MarkMeasurementError(newMeasurement, err)
+		}
+		jobId := dat["canaryExecutionId"]
+		m := make(map[string]string)
+		m["canaryExecutionId"] = fmt.Sprintf("%v", jobId)
+		if len(m["canaryExecutionId"]) == 0 {
+
+			return metricutil.MarkMeasurementError(newMeasurement, errors.New("Invalid canaryExecutionId"))
+		}
+		newMeasurement.Metadata = m
+	}
+
+	newMeasurement.Phase = v1alpha1.AnalysisPhaseRunning
+
+	resumeTime := metav1.NewTime(time.Now().Add(15 * time.Second))
+	newMeasurement.ResumeAt = &resumeTime
+	finishTime := metav1.Now()
+	newMeasurement.FinishedAt = &finishTime
+
+	return newMeasurement
+}
+
+// Resume should not be used the kayenta provider since all the work should occur in the Run method
+func (p *Provider) Resume(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric, measurement v1alpha1.Measurement) v1alpha1.Measurement {
+
+	scoreURL := strings.Replace(KayentaScoreURL, "{{inputs.address}}", metric.Provider.Kayenta.Address, 1)
+	scoreURL = strings.Replace(scoreURL, "{{inputs.canaryExecutionId}}", measurement.Metadata["canaryExecutionId"], 1)
+
+	response, err := p.client.Get(scoreURL)
+	if err != nil || response.Body == nil || response.StatusCode > 300 {
+		if err == nil {
+			err := errors.New("Invalid Response")
+			return metricutil.MarkMeasurementError(measurement, err)
+		}
+		return metricutil.MarkMeasurementError(measurement, err)
+	} else {
+		data, err := ioutil.ReadAll(response.Body)
+		if  err != nil {
+			return metricutil.MarkMeasurementError(measurement, err)
+		}
+		json := string(data)
+		result := gjson.Get(json, "result.judgeResult.score.score")
```

Comment:
Can you use something like this instead https://github.com/argoproj/argo-rollouts/blob/d32942a163b50c4b9daa122cbbe3ad37464f87de/rollout/canary_test.go#L1005 so we do not have to introduce a new library?

Reply:
The library has a similar function called NestedInt64 that instead returns an int so I think would be a better fit